v12.5.0 proposal

BridgeAR opened the pull request:
Notable changes:

**TBD**

Commits:

* [[`f03241fc0a`](https://github.com/nodejs/node/commit/f03241fc0a)] - **(SEMVER-MINOR)** **assert**: add partial support for evaluated code in simple assert (Ruben Bridgewater) [#27781](https://github.com/nodejs/node/pull/27781)
* [[`ef8f147b7e`](https://github.com/nodejs/node/commit/ef8f147b7e)] - **(SEMVER-MINOR)** **assert**: improve regular expression validation (Ruben Bridgewater) [#27781](https://github.com/nodejs/node/pull/27781)
* [[`8157a50161`](https://github.com/nodejs/node/commit/8157a50161)] - **assert**: print more lines in the error diff (Ruben Bridgewater) [#28058](https://github.com/nodejs/node/pull/28058)
* [[`82174412a5`](https://github.com/nodejs/node/commit/82174412a5)] - **assert**: fix error diff (Ruben Bridgewater) [#28058](https://github.com/nodejs/node/pull/28058)
* [[`1ee7ce6092`](https://github.com/nodejs/node/commit/1ee7ce6092)] - **assert**: limit string inspection when logging assertion errors (Ruben Bridgewater) [#28058](https://github.com/nodejs/node/pull/28058)
* [[`366c9831cd`](https://github.com/nodejs/node/commit/366c9831cd)] - **build**: fix cctest target for --without-report (Richard Lau) [#28238](https://github.com/nodejs/node/pull/28238)
* [[`47dc5b2155`](https://github.com/nodejs/node/commit/47dc5b2155)] - **build**: guard test-doc recipe with node\_use\_openssl (Daniel Bevenius) [#28199](https://github.com/nodejs/node/pull/28199)
* [[`2201484541`](https://github.com/nodejs/node/commit/2201484541)] - **build**: turn on custom V8 snapshot by default (Joyee Cheung) [#28181](https://github.com/nodejs/node/pull/28181)
* [[`7d0ffee435`](https://github.com/nodejs/node/commit/7d0ffee435)] - **build**: unbreak --with-intl=system-icu build (Ben Noordhuis) [#28118](https://github.com/nodejs/node/pull/28118)
* [[`0931f7e635`](https://github.com/nodejs/node/commit/0931f7e635)] - **build**: fix icu-i18n pkg-config version check (Ben Noordhuis) [#28118](https://github.com/nodejs/node/pull/28118)
* [[`e2ad36482c`](https://github.com/nodejs/node/commit/e2ad36482c)] - **build**: don't swallow pkg-config warnings (Ben Noordhuis) [#28118](https://github.com/nodejs/node/pull/28118)
* [[`48d7d7c53e`](https://github.com/nodejs/node/commit/48d7d7c53e)] - **build**: lint all docs under doc (Richard Lau) [#28128](https://github.com/nodejs/node/pull/28128)
* [[`d3207912fb`](https://github.com/nodejs/node/commit/d3207912fb)] - **build**: fix configure script to work with Apple Clang 11 (Saagar Jha) [#28071](https://github.com/nodejs/node/pull/28071)
* [[`21bcfb67f6`](https://github.com/nodejs/node/commit/21bcfb67f6)] - **(SEMVER-MAJOR)** **build**: reset embedder string to "-node.0" (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`e5c26753e9`](https://github.com/nodejs/node/commit/e5c26753e9)] - **build,meta**: rearrange and narrow git ignore rules (Refael Ackermann) [#27954](https://github.com/nodejs/node/pull/27954)
* [[`5101e4c2a2`](https://github.com/nodejs/node/commit/5101e4c2a2)] - **(SEMVER-MAJOR)** **build,v8**: sync V8 gypfiles with 7.5 (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`5a7154ef32`](https://github.com/nodejs/node/commit/5a7154ef32)] - **build,win**: delegate lint-cpp to make (Refael Ackermann) [#28102](https://github.com/nodejs/node/pull/28102)
* [[`3f1787b47d`](https://github.com/nodejs/node/commit/3f1787b47d)] - **crypto**: add debug info client emit secureConnect (Daniel Bevenius) [#28067](https://github.com/nodejs/node/pull/28067)
* [[`5fddde69bb`](https://github.com/nodejs/node/commit/5fddde69bb)] - **deps**: update archs files for OpenSSL-1.1.1c (Sam Roberts) [#28211](https://github.com/nodejs/node/pull/28211)
* [[`0a2353eb9a`](https://github.com/nodejs/node/commit/0a2353eb9a)] - **deps**: upgrade openssl sources to 1.1.1c (Sam Roberts) [#28211](https://github.com/nodejs/node/pull/28211)
* [[`e46d9e9df7`](https://github.com/nodejs/node/commit/e46d9e9df7)] - **deps**: updated openssl upgrade instructions (Sam Roberts) [#28211](https://github.com/nodejs/node/pull/28211)
* [[`084ffd8c2f`](https://github.com/nodejs/node/commit/084ffd8c2f)] - **deps**: update llhttp to 1.1.4 (Fedor Indutny) [#28154](https://github.com/nodejs/node/pull/28154)
* [[`9382b3be9c`](https://github.com/nodejs/node/commit/9382b3be9c)] - **deps**: V8: cherry-pick e0a109c (Joyee Cheung) [#27533](https://github.com/nodejs/node/pull/27533)
* [[`b690e19a9a`](https://github.com/nodejs/node/commit/b690e19a9a)] - **deps**: ignore deps/.cipd fetched by deps/v8/tools/node/fetch\_deps.py (Joyee Cheung) [#28095](https://github.com/nodejs/node/pull/28095)
* [[`d42ad64253`](https://github.com/nodejs/node/commit/d42ad64253)] - **deps**: update node-inspect to v1.11.6 (Jan Krems) [#28039](https://github.com/nodejs/node/pull/28039)
* [[`40a1a11542`](https://github.com/nodejs/node/commit/40a1a11542)] - **(SEMVER-MINOR)** **deps**: patch V8 to be API/ABI compatible with 7.4 (Michaël Zasso) [#28005](https://github.com/nodejs/node/pull/28005)
* [[`ad3a164ec3`](https://github.com/nodejs/node/commit/ad3a164ec3)] - **(SEMVER-MAJOR)** **deps**: bump minimum icu version to 64 (Michaël Zasso) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`e4aa869726`](https://github.com/nodejs/node/commit/e4aa869726)] - **(SEMVER-MAJOR)** **deps**: V8: backport 3a75c1f (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`bb729a415a`](https://github.com/nodejs/node/commit/bb729a415a)] - **(SEMVER-MAJOR)** **deps**: V8: fix BUILDING\_V8\_SHARED issues (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`f8a33abe0c`](https://github.com/nodejs/node/commit/f8a33abe0c)] - **(SEMVER-MAJOR)** **deps**: V8: workaround for MSVC 14.20 optimizer bug (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`0a5ff4cb33`](https://github.com/nodejs/node/commit/0a5ff4cb33)] - **(SEMVER-MAJOR)** **deps**: V8: template explicit instantiation for GCC-8 (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`b411114a52`](https://github.com/nodejs/node/commit/b411114a52)] - **(SEMVER-MAJOR)** **deps**: V8: use ATOMIC\_VAR\_INIT instead of std::atomic\_init (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`c08d94baef`](https://github.com/nodejs/node/commit/c08d94baef)] - **(SEMVER-MAJOR)** **deps**: V8: forward declaration of `Rtl\*FunctionTable` (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`445bb81ab6`](https://github.com/nodejs/node/commit/445bb81ab6)] - **(SEMVER-MAJOR)** **deps**: V8: patch register-arm64.h (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`fa6dfec186`](https://github.com/nodejs/node/commit/fa6dfec186)] - **(SEMVER-MAJOR)** **deps**: V8: backport f89e555 (Michaël Zasso) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`8b8fe87e54`](https://github.com/nodejs/node/commit/8b8fe87e54)] - **deps**: V8: cherry-pick cca9ae3c9a (Benedikt Meurer) [#27729](https://github.com/nodejs/node/pull/27729)
* [[`55e99448c8`](https://github.com/nodejs/node/commit/55e99448c8)] - **(SEMVER-MAJOR)** **deps**: V8: update postmortem metadata generation script (cjihrig) [#26685](https://github.com/nodejs/node/pull/26685)
* [[`2f92b15435`](https://github.com/nodejs/node/commit/2f92b15435)] - **(SEMVER-MAJOR)** **deps**: V8: silence irrelevant warning (Michaël Zasso) [#26685](https://github.com/nodejs/node/pull/26685)
* [[`ca8e5aa77b`](https://github.com/nodejs/node/commit/ca8e5aa77b)] - **(SEMVER-MAJOR)** **deps**: V8: un-cherry-pick bd019bd (Refael Ackermann) [#26685](https://github.com/nodejs/node/pull/26685)
* [[`4a61bdbc1f`](https://github.com/nodejs/node/commit/4a61bdbc1f)] - **(SEMVER-MAJOR)** **deps**: V8: fix filename manipulation for Windows (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`86a8bb7612`](https://github.com/nodejs/node/commit/86a8bb7612)] - **(SEMVER-MAJOR)** **deps**: update V8 to 7.5.288.22 (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`071694472f`](https://github.com/nodejs/node/commit/071694472f)] - **dgram**: fix abort on bad args (cjihrig) [#28135](https://github.com/nodejs/node/pull/28135)
* [[`ed8b9dc51d`](https://github.com/nodejs/node/commit/ed8b9dc51d)] - **doc**: revise intro sentence for assert (Rich Trott) [#28226](https://github.com/nodejs/node/pull/28226)
* [[`5b7b18d4df`](https://github.com/nodejs/node/commit/5b7b18d4df)] - **doc**: improve assert strict-mode text (Rich Trott) [#28239](https://github.com/nodejs/node/pull/28239)
* [[`833d3d5ad1`](https://github.com/nodejs/node/commit/833d3d5ad1)] - **doc**: clarify commit message format in pull-requests.md (rexagod) [#28125](https://github.com/nodejs/node/pull/28125)
* [[`f6b97f2aed`](https://github.com/nodejs/node/commit/f6b97f2aed)] - **doc**: add missing options allowed in NODE\_OPTIONS (Richard Lau) [#28179](https://github.com/nodejs/node/pull/28179)
* [[`37806839cf`](https://github.com/nodejs/node/commit/37806839cf)] - **doc**: document behavior of family:0 in dns.lookup() (cjihrig) [#28163](https://github.com/nodejs/node/pull/28163)
* [[`07ab5917f7`](https://github.com/nodejs/node/commit/07ab5917f7)] - **doc**: pass path in URL constructor (Daniel Nalborczyk) [#28161](https://github.com/nodejs/node/pull/28161)
* [[`41af9bfb24`](https://github.com/nodejs/node/commit/41af9bfb24)] - **doc**: update kernel and glibc reqs for PPCle (Michael Dawson) [#28162](https://github.com/nodejs/node/pull/28162)
* [[`abe5d05523`](https://github.com/nodejs/node/commit/abe5d05523)] - **(SEMVER-MINOR)** **doc**: update assert's validation functions (Ruben Bridgewater) [#27781](https://github.com/nodejs/node/pull/27781)
* [[`ecb963dd44`](https://github.com/nodejs/node/commit/ecb963dd44)] - **doc**: document trace-events category for dns requests (vmarchaud) [#28100](https://github.com/nodejs/node/pull/28100)
* [[`8c277555b7`](https://github.com/nodejs/node/commit/8c277555b7)] - **doc**: add Buffer#subarray() and add note about Uint8Array#slice() (FUJI Goro (gfx)) [#28101](https://github.com/nodejs/node/pull/28101)
* [[`4d6262fb56`](https://github.com/nodejs/node/commit/4d6262fb56)] - **doc**: update broken/foundation links in README.md (Tierney Cyren) [#28119](https://github.com/nodejs/node/pull/28119)
* [[`00e6c9d2dd`](https://github.com/nodejs/node/commit/00e6c9d2dd)] - **doc**: add tls-min/max options to NODE\_OPTIONS (Daniel Bevenius) [#28146](https://github.com/nodejs/node/pull/28146)
* [[`705f259142`](https://github.com/nodejs/node/commit/705f259142)] - **doc**: split example into two (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`69af43ead9`](https://github.com/nodejs/node/commit/69af43ead9)] - **doc**: clarify N-API version Matrix (Michael Dawson) [#27942](https://github.com/nodejs/node/pull/27942)
* [[`56b150b3d7`](https://github.com/nodejs/node/commit/56b150b3d7)] - **doc**: add current recommendation for ESM/CommonJS dual packages (Geoffrey Booth) [#27957](https://github.com/nodejs/node/pull/27957)
* [[`360c708f64`](https://github.com/nodejs/node/commit/360c708f64)] - **doc**: document Http2Stream#id property (murgatroid99) [#28074](https://github.com/nodejs/node/pull/28074)
* [[`5fc4e48fdd`](https://github.com/nodejs/node/commit/5fc4e48fdd)] - **doc**: add note about AsyncResource for Worker pooling (Anna Henningsen) [#28023](https://github.com/nodejs/node/pull/28023)
* [[`d1c53fc54e`](https://github.com/nodejs/node/commit/d1c53fc54e)] - **doc**: fix `prohibited-strings` warning in `pull-requests.md` (rexagod) [#28127](https://github.com/nodejs/node/pull/28127)
* [[`e6ecc13cfa`](https://github.com/nodejs/node/commit/e6ecc13cfa)] - **doc**: improve synopsis.md (Rich Trott) [#28115](https://github.com/nodejs/node/pull/28115)
* [[`eb05db907a`](https://github.com/nodejs/node/commit/eb05db907a)] - **doc**: edit reason-for-deprecation text (Rich Trott) [#28098](https://github.com/nodejs/node/pull/28098)
* [[`5ad0d047c6`](https://github.com/nodejs/node/commit/5ad0d047c6)] - **doc**: improve DEP0090 text (Rich Trott) [#28097](https://github.com/nodejs/node/pull/28097)
* [[`9074f9b4e5`](https://github.com/nodejs/node/commit/9074f9b4e5)] - **doc**: clarify special schemes (Rich Trott) [#28091](https://github.com/nodejs/node/pull/28091)
* [[`f95a52cb1e`](https://github.com/nodejs/node/commit/f95a52cb1e)] - **doc**: clarify weak keys text (Rich Trott) [#28090](https://github.com/nodejs/node/pull/28090)
* [[`eb73ed8158`](https://github.com/nodejs/node/commit/eb73ed8158)] - **doc**: remove superfluous filenaming convention (Rich Trott) [#28089](https://github.com/nodejs/node/pull/28089)
* [[`f7d8384af2`](https://github.com/nodejs/node/commit/f7d8384af2)] - **doc**: mark Node.js 11 as EOL in changelog (Richard Lau) [#28076](https://github.com/nodejs/node/pull/28076)
* [[`87c55ea0ef`](https://github.com/nodejs/node/commit/87c55ea0ef)] - **doc**: adjust TOC margins (Roman Reiss) [#28075](https://github.com/nodejs/node/pull/28075)
* [[`9dd4813008`](https://github.com/nodejs/node/commit/9dd4813008)] - **doc**: order deprecation reasons (Trivikram Kamat) [#27960](https://github.com/nodejs/node/pull/27960)
* [[`e3f905ac7e`](https://github.com/nodejs/node/commit/e3f905ac7e)] - **doc**: remove "encouraged" as hedging in fs.md (Rich Trott) [#28027](https://github.com/nodejs/node/pull/28027)
* [[`df22b96cb0`](https://github.com/nodejs/node/commit/df22b96cb0)] - **doc**: remove "strongly recommended" as hedging in fs.md (Rich Trott) [#28028](https://github.com/nodejs/node/pull/28028)
* [[`049429bd97`](https://github.com/nodejs/node/commit/049429bd97)] - **doc**: remove "strongly recommended" hedging from tls.md (Rich Trott) [#28029](https://github.com/nodejs/node/pull/28029)
* [[`79d4f285be`](https://github.com/nodejs/node/commit/79d4f285be)] - **doc**: remove "strongly recommended" hedging in deprecations.md (Rich Trott) [#28031](https://github.com/nodejs/node/pull/28031)
* [[`e36ed72855`](https://github.com/nodejs/node/commit/e36ed72855)] - **doc,n-api**: fix typo (Richard Lau) [#28178](https://github.com/nodejs/node/pull/28178)
* [[`c574b57d65`](https://github.com/nodejs/node/commit/c574b57d65)] - **doc,test**: test documentation consistency for NODE\_OPTIONS (Richard Lau) [#28179](https://github.com/nodejs/node/pull/28179)
* [[`e89a39bbb7`](https://github.com/nodejs/node/commit/e89a39bbb7)] - **fs**: document the Date conversion in Stats objects (Joyee Cheung) [#28224](https://github.com/nodejs/node/pull/28224)
* [[`d36adcdf34`](https://github.com/nodejs/node/commit/d36adcdf34)] - **(SEMVER-MINOR)** **fs**: add \*timeNs properties to BigInt Stats objects (Joyee Cheung) [#21387](https://github.com/nodejs/node/pull/21387)
* [[`e1fc9b987a`](https://github.com/nodejs/node/commit/e1fc9b987a)] - **http2**: do not register unnecessary listeners (Antonio Kukas) [#27987](https://github.com/nodejs/node/pull/27987)
* [[`53cc16c7dd`](https://github.com/nodejs/node/commit/53cc16c7dd)] - **https**: do not automatically use invalid servername (Sam Roberts) [#28209](https://github.com/nodejs/node/pull/28209)
* [[`f8c9a58bf5`](https://github.com/nodejs/node/commit/f8c9a58bf5)] - **inspector**: added --inspect-publish-uid (Aleksei Koziatinskii) [#27741](https://github.com/nodejs/node/pull/27741)
* [[`ae7789a3c1`](https://github.com/nodejs/node/commit/ae7789a3c1)] - **module**: prevent race condition while combining import and require (Ruben Bridgewater) [#27674](https://github.com/nodejs/node/pull/27674)
* [[`6014429580`](https://github.com/nodejs/node/commit/6014429580)] - **module**: handle empty require.resolve() options (cjihrig) [#28078](https://github.com/nodejs/node/pull/28078)
* [[`83fc10dede`](https://github.com/nodejs/node/commit/83fc10dede)] - **n-api**: define ECMAScript-compliant accessors on napi\_define\_class (legendecas) [#27851](https://github.com/nodejs/node/pull/27851)
* [[`d70fd7dddd`](https://github.com/nodejs/node/commit/d70fd7dddd)] - **n-api**: define ECMAScript-compliant accessors on napi\_define\_properties (legendecas) [#27851](https://github.com/nodejs/node/pull/27851)
* [[`39063b3c00`](https://github.com/nodejs/node/commit/39063b3c00)] - **n-api**: defer Buffer finalizer with SetImmediate (Anna Henningsen) [#28082](https://github.com/nodejs/node/pull/28082)
* [[`dfbbfbb765`](https://github.com/nodejs/node/commit/dfbbfbb765)] - **net**: make writeAfterFIN() return false (Luigi Pinca) [#27996](https://github.com/nodejs/node/pull/27996)
* [[`44b0e13b1c`](https://github.com/nodejs/node/commit/44b0e13b1c)] - **perf_hooks,trace_events**: use stricter equality (cjihrig) [#28166](https://github.com/nodejs/node/pull/28166)
* [[`50c803828f`](https://github.com/nodejs/node/commit/50c803828f)] - **process**: refactor unhandled rejection handling (Joyee Cheung) [#28228](https://github.com/nodejs/node/pull/28228)
* [[`b491eabff1`](https://github.com/nodejs/node/commit/b491eabff1)] - **process**: improve queueMicrotask performance (Anatoli Papirovski) [#28093](https://github.com/nodejs/node/pull/28093)
* [[`460cc6285a`](https://github.com/nodejs/node/commit/460cc6285a)] - **process**: code cleanup for nextTick (Anatoli Papirovski) [#28047](https://github.com/nodejs/node/pull/28047)
* [[`682105538a`](https://github.com/nodejs/node/commit/682105538a)] - **report**: add cpu info to report output (Christopher Hiller) [#28188](https://github.com/nodejs/node/pull/28188)
* [[`7a4a94a54f`](https://github.com/nodejs/node/commit/7a4a94a54f)] - **src**: fix compiler warning in node\_worker.cc (Daniel Bevenius) [#28198](https://github.com/nodejs/node/pull/28198)
* [[`b9d21e5bfd`](https://github.com/nodejs/node/commit/b9d21e5bfd)] - **src**: fix off-by-one error in native SetImmediate (Anna Henningsen) [#28082](https://github.com/nodejs/node/pull/28082)
* [[`c67642ae03`](https://github.com/nodejs/node/commit/c67642ae03)] - **src**: do not use pointer for loop in node\_watchdog (Anna Henningsen) [#28020](https://github.com/nodejs/node/pull/28020)
* [[`b5dda32b8a`](https://github.com/nodejs/node/commit/b5dda32b8a)] - **src**: restore stdio on program exit (Ben Noordhuis) [#24260](https://github.com/nodejs/node/pull/24260)
* [[`8984b73033`](https://github.com/nodejs/node/commit/8984b73033)] - **src**: remove TLS code for unsupported OpenSSLs (Sam Roberts) [#28085](https://github.com/nodejs/node/pull/28085)
* [[`8849eb24c1`](https://github.com/nodejs/node/commit/8849eb24c1)] - **src**: handle exceptions from ToDetailString() (Anna Henningsen) [#28019](https://github.com/nodejs/node/pull/28019)
* [[`8a032fc50c`](https://github.com/nodejs/node/commit/8a032fc50c)] - **src**: expose DOMException to internalBinding('message') for testing (Joyee Cheung) [#28072](https://github.com/nodejs/node/pull/28072)
* [[`a5fdedb3d5`](https://github.com/nodejs/node/commit/a5fdedb3d5)] - **src**: only run preloadModules if the preload array is not empty (Samuel Attard) [#28012](https://github.com/nodejs/node/pull/28012)
* [[`c821eefa5f`](https://github.com/nodejs/node/commit/c821eefa5f)] - **src**: add napi\_define\_class() null checks (Octavian Soldea) [#27945](https://github.com/nodejs/node/pull/27945)
* [[`95ee3b55d3`](https://github.com/nodejs/node/commit/95ee3b55d3)] - **src**: use RAII in setgroups implementation (Anna Henningsen) [#28022](https://github.com/nodejs/node/pull/28022)
* [[`d81c67bd8f`](https://github.com/nodejs/node/commit/d81c67bd8f)] - **src**: fix unused private field warning (cjihrig) [#28036](https://github.com/nodejs/node/pull/28036)
* [[`e8bedd2009`](https://github.com/nodejs/node/commit/e8bedd2009)] - **src**: split `RunBootstrapping()` (Joyee Cheung) [#27539](https://github.com/nodejs/node/pull/27539)
* [[`c20c6e55b5`](https://github.com/nodejs/node/commit/c20c6e55b5)] - **src**: reorganize inspector and diagnostics initialization (Joyee Cheung) [#27539](https://github.com/nodejs/node/pull/27539)
* [[`c086736a49`](https://github.com/nodejs/node/commit/c086736a49)] - **src**: create Environment properties in Environment::CreateProperties() (Joyee Cheung) [#27539](https://github.com/nodejs/node/pull/27539)
* [[`70f8e71a0d`](https://github.com/nodejs/node/commit/70f8e71a0d)] - **src**: inline ProcessCliArgs in the Environment constructor (Joyee Cheung) [#27539](https://github.com/nodejs/node/pull/27539)
* [[`edb1a3c3c1`](https://github.com/nodejs/node/commit/edb1a3c3c1)] - **test**: add eval ESM module tests (Evgenii Shchepotev) [#27956](https://github.com/nodejs/node/pull/27956)
* [[`42f5f27727`](https://github.com/nodejs/node/commit/42f5f27727)] - **test**: fix NODE\_OPTIONS feature check (Richard Lau) [#28225](https://github.com/nodejs/node/pull/28225)
* [[`720a2a6da3`](https://github.com/nodejs/node/commit/720a2a6da3)] - **test**: move --cpu-prof tests to sequential (Joyee Cheung) [#28210](https://github.com/nodejs/node/pull/28210)
* [[`6aa64517e5`](https://github.com/nodejs/node/commit/6aa64517e5)] - **test**: skip `test-worker-prof` as flaky for all (Milad Farazmand) [#28175](https://github.com/nodejs/node/pull/28175)
* [[`414e9d2811`](https://github.com/nodejs/node/commit/414e9d2811)] - **test**: remove FIB environment variable from cpu-prof.js (Rich Trott) [#28183](https://github.com/nodejs/node/pull/28183)
* [[`066d814b3c`](https://github.com/nodejs/node/commit/066d814b3c)] - **test**: remove unused `output` argument for `getFrames()` (Rich Trott) [#28183](https://github.com/nodejs/node/pull/28183)
* [[`5d8dce41e3`](https://github.com/nodejs/node/commit/5d8dce41e3)] - **test**: document cpu-prof module (Rich Trott) [#28183](https://github.com/nodejs/node/pull/28183)
* [[`318328f6b7`](https://github.com/nodejs/node/commit/318328f6b7)] - **test**: improve unexpected warnings error (Ruben Bridgewater) [#28138](https://github.com/nodejs/node/pull/28138)
* [[`31ccd36668`](https://github.com/nodejs/node/commit/31ccd36668)] - **test**: mark test-fs-stat-bigint as flaky (Rich Trott) [#28156](https://github.com/nodejs/node/pull/28156)
* [[`de6627f9a4`](https://github.com/nodejs/node/commit/de6627f9a4)] - **test**: remove duplicate test-child-process-execfilesync-maxBuffer.js (Joyee Cheung) [#28139](https://github.com/nodejs/node/pull/28139)
* [[`2353c63dc2`](https://github.com/nodejs/node/commit/2353c63dc2)] - **test**: split test-cpu-prof.js (Joyee Cheung) [#28170](https://github.com/nodejs/node/pull/28170)
* [[`186e94c322`](https://github.com/nodejs/node/commit/186e94c322)] - **test**: add github refs to flaky tests (Sam Roberts) [#28123](https://github.com/nodejs/node/pull/28123)
* [[`d8061dc1a0`](https://github.com/nodejs/node/commit/d8061dc1a0)] - **test**: remove test-gc-http-client from status file (Rich Trott) [#28130](https://github.com/nodejs/node/pull/28130)
* [[`d6791d1cb8`](https://github.com/nodejs/node/commit/d6791d1cb8)] - **test**: remove test-tty-wrap from status file (Rich Trott) [#28129](https://github.com/nodejs/node/pull/28129)
* [[`b0bc23c572`](https://github.com/nodejs/node/commit/b0bc23c572)] - **test**: add comments to the foaf+ssl fixtures (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`189d6af2b3`](https://github.com/nodejs/node/commit/189d6af2b3)] - **test**: change formatting of fixtures/keys/Makefile (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`94a6d7a518`](https://github.com/nodejs/node/commit/94a6d7a518)] - **test**: change fixtures.readSync to fixtures.readKey (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`c82023a173`](https://github.com/nodejs/node/commit/c82023a173)] - **test**: remove uneeded agent keypair in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`74e6109f39`](https://github.com/nodejs/node/commit/74e6109f39)] - **test**: move foafssl certs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`78f39c91ac`](https://github.com/nodejs/node/commit/78f39c91ac)] - **test**: remove uneeded alice certs in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`5d0737735b`](https://github.com/nodejs/node/commit/5d0737735b)] - **test**: remove uneeded certs in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`d757e0b6d4`](https://github.com/nodejs/node/commit/d757e0b6d4)] - **test**: move dherror.pem to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`f9ddcc6305`](https://github.com/nodejs/node/commit/f9ddcc6305)] - **test**: remove pass-\* certs (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`e673b57055`](https://github.com/nodejs/node/commit/e673b57055)] - **test**: move test\_\[key|ca|cert\] to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`8670f6dd22`](https://github.com/nodejs/node/commit/8670f6dd22)] - **test**: move spkac certs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`7d1f15fba1`](https://github.com/nodejs/node/commit/7d1f15fba1)] - **test**: move x448 keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`22bbdc5068`](https://github.com/nodejs/node/commit/22bbdc5068)] - **test**: move ed448 keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`9de9d55bfc`](https://github.com/nodejs/node/commit/9de9d55bfc)] - **test**: move dsa keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`9684842023`](https://github.com/nodejs/node/commit/9684842023)] - **test**: move rsa keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`7ae23abc01`](https://github.com/nodejs/node/commit/7ae23abc01)] - **test**: move x25519 keypair to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`adb0197d6d`](https://github.com/nodejs/node/commit/adb0197d6d)] - **test**: move ed25519 keypair to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`14bd26c8da`](https://github.com/nodejs/node/commit/14bd26c8da)] - **test**: remove workaround for unsupported OpenSSLs (Sam Roberts) [#28085](https://github.com/nodejs/node/pull/28085)
* [[`d4bb88eed8`](https://github.com/nodejs/node/commit/d4bb88eed8)] - **test**: simplify tests code (himself65) [#28065](https://github.com/nodejs/node/pull/28065)
* [[`87e977ae42`](https://github.com/nodejs/node/commit/87e977ae42)] - **test**: make sure vtable is generated in addon test with LTO (Anna Henningsen) [#28057](https://github.com/nodejs/node/pull/28057)
* [[`3feaf3ddbe`](https://github.com/nodejs/node/commit/3feaf3ddbe)] - **test**: mark test-worker-debug as flaky (Refael Ackermann) [#28035](https://github.com/nodejs/node/pull/28035)
* [[`098cf74292`](https://github.com/nodejs/node/commit/098cf74292)] - **test**: regression test `tmpdir` (Refael Ackermann) [#28035](https://github.com/nodejs/node/pull/28035)
* [[`e08a98fa43`](https://github.com/nodejs/node/commit/e08a98fa43)] - **test**: always suffix `tmpdir` (Refael Ackermann) [#28035](https://github.com/nodejs/node/pull/28035)
* [[`f33623662f`](https://github.com/nodejs/node/commit/f33623662f)] - **test**: shell out to `rmdir` first on Windows (Refael Ackermann) [#28035](https://github.com/nodejs/node/pull/28035)
* [[`1ef2811236`](https://github.com/nodejs/node/commit/1ef2811236)] - **test**: only assert on first lines of TLS trace (Sam Roberts) [#28043](https://github.com/nodejs/node/pull/28043)
* [[`62de36e8d3`](https://github.com/nodejs/node/commit/62de36e8d3)] - ***Revert*** "**test**: move all test keys/certs under `test/fixtures/keys/`" (Sam Roberts) [#28083](https://github.com/nodejs/node/pull/28083)
* [[`2331e9c380`](https://github.com/nodejs/node/commit/2331e9c380)] - **test**: add comments to the foaf+ssl fixtures (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`8e28259bf8`](https://github.com/nodejs/node/commit/8e28259bf8)] - **test**: change formatting of fixtures/keys/Makefile (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`d258504a31`](https://github.com/nodejs/node/commit/d258504a31)] - **test**: change fixtures.readSync to fixtures.readKey (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`328b2d0c88`](https://github.com/nodejs/node/commit/328b2d0c88)] - **test**: remove uneeded agent keypair in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`a0d2862b1e`](https://github.com/nodejs/node/commit/a0d2862b1e)] - **test**: move foafssl certs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`af9eb9648e`](https://github.com/nodejs/node/commit/af9eb9648e)] - **test**: remove uneeded alice certs in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`ee62fa172c`](https://github.com/nodejs/node/commit/ee62fa172c)] - **test**: remove uneeded certs in fixtures/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`f41dfd71a0`](https://github.com/nodejs/node/commit/f41dfd71a0)] - **test**: move dherror.pem to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`98f7ae9e7b`](https://github.com/nodejs/node/commit/98f7ae9e7b)] - **test**: remove pass-\* certs (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`27d6b28943`](https://github.com/nodejs/node/commit/27d6b28943)] - **test**: move test\_\[key|ca|cert\] to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`5ac6dddb83`](https://github.com/nodejs/node/commit/5ac6dddb83)] - **test**: move spkac certs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`03b92e93b1`](https://github.com/nodejs/node/commit/03b92e93b1)] - **test**: move x448 keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`4155bbaeab`](https://github.com/nodejs/node/commit/4155bbaeab)] - **test**: move ed448 keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`9209698139`](https://github.com/nodejs/node/commit/9209698139)] - **test**: move dsa keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`ad42258d5a`](https://github.com/nodejs/node/commit/ad42258d5a)] - **test**: move rsa keypairs to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`686cb13f78`](https://github.com/nodejs/node/commit/686cb13f78)] - **test**: move x25519 keypair to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`1f2de2fbe1`](https://github.com/nodejs/node/commit/1f2de2fbe1)] - **test**: move ed25519 keypair to fixtures/keys/ (Alex Aubuchon) [#27962](https://github.com/nodejs/node/pull/27962)
* [[`687e57fe19`](https://github.com/nodejs/node/commit/687e57fe19)] - **test**: rename worker MessagePort test (Anna Henningsen) [#28024](https://github.com/nodejs/node/pull/28024)
* [[`7165254f8b`](https://github.com/nodejs/node/commit/7165254f8b)] - **test**: more tls hostname verification coverage (Ben Noordhuis) [#27999](https://github.com/nodejs/node/pull/27999)
* [[`92d1ca9645`](https://github.com/nodejs/node/commit/92d1ca9645)] - **(SEMVER-MAJOR)** **test**: fail `test-worker-prof` on internal timeout (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`ab1a4eb12d`](https://github.com/nodejs/node/commit/ab1a4eb12d)] - **(SEMVER-MAJOR)** **test**: drain platform before unregistering isolate (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`b6bdf752d6`](https://github.com/nodejs/node/commit/b6bdf752d6)] - **test,v8**: skip less and stabilize test-linux-perf.js (Refael Ackermann) [#27364](https://github.com/nodejs/node/pull/27364)
* [[`e270ff1166`](https://github.com/nodejs/node/commit/e270ff1166)] - **tls**: remove unnecessary set of DEFAULT\_MAX\_VERSION (Daniel Bevenius) [#28147](https://github.com/nodejs/node/pull/28147)
* [[`6b9d477520`](https://github.com/nodejs/node/commit/6b9d477520)] - **tls**: rename validateKeyCert in \_tls\_common.js (Daniel Bevenius) [#28116](https://github.com/nodejs/node/pull/28116)
* [[`90854124dd`](https://github.com/nodejs/node/commit/90854124dd)] - **tools**: assert that the snapshot can be rehashed in node\_mksnapshot (Joyee Cheung) [#28181](https://github.com/nodejs/node/pull/28181)
* [[`a0c5b58b44`](https://github.com/nodejs/node/commit/a0c5b58b44)] - **tools**: fix update-babel-eslint.sh script (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`6460d071d2`](https://github.com/nodejs/node/commit/6460d071d2)] - **tools**: increase the maximum number of files to lint per worker (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`bf76823a47`](https://github.com/nodejs/node/commit/bf76823a47)] - **tools**: ignore node\_modules when linting (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`52564dbb28`](https://github.com/nodejs/node/commit/52564dbb28)] - **tools**: update babel-eslint (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`9be67b2cbc`](https://github.com/nodejs/node/commit/9be67b2cbc)] - **tools**: activate more eslint rules (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`739c2a3336`](https://github.com/nodejs/node/commit/739c2a3336)] - **tools**: update eslint config (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`adfbd362fc`](https://github.com/nodejs/node/commit/adfbd362fc)] - **tools**: update eslint (Ruben Bridgewater) [#27670](https://github.com/nodejs/node/pull/27670)
* [[`018159d734`](https://github.com/nodejs/node/commit/018159d734)] - **(SEMVER-MAJOR)** **tools,gyp**: introduce MSVS 2019 (Refael Ackermann) [#27375](https://github.com/nodejs/node/pull/27375)
* [[`9dd840dff5`](https://github.com/nodejs/node/commit/9dd840dff5)] - **trace_events**: respect inspect() depth (cjihrig) [#28037](https://github.com/nodejs/node/pull/28037)
* [[`225fb1bff7`](https://github.com/nodejs/node/commit/225fb1bff7)] - **util**: use average bias while grouping arrays (Ruben Bridgewater) [#28070](https://github.com/nodejs/node/pull/28070)
* [[`8f4bb1949f`](https://github.com/nodejs/node/commit/8f4bb1949f)] - **util**: improve .inspect() array grouping (Ruben Bridgewater) [#28070](https://github.com/nodejs/node/pull/28070)
* [[`9ced334a6c`](https://github.com/nodejs/node/commit/9ced334a6c)] - **util**: refactor inspecting long lines (Ruben Bridgewater) [#28055](https://github.com/nodejs/node/pull/28055)
* [[`dfdf742fd1`](https://github.com/nodejs/node/commit/dfdf742fd1)] - **util**: use Set to store deprecation codes (Daniel Nalborczyk) [#28113](https://github.com/nodejs/node/pull/28113)
* [[`c3243de47a`](https://github.com/nodejs/node/commit/c3243de47a)] - **util**: special handle `maxArrayLength` while grouping arrays (Ruben Bridgewater) [#28059](https://github.com/nodejs/node/pull/28059)
* [[`f897860427`](https://github.com/nodejs/node/commit/f897860427)] - **util**: support AsyncGeneratorFunction in .inspect (Ruben Bridgewater) [#28056](https://github.com/nodejs/node/pull/28056)
* [[`3611b471f7`](https://github.com/nodejs/node/commit/3611b471f7)] - **worker**: refactor `worker.terminate()` (Anna Henningsen) [#28021](https://github.com/nodejs/node/pull/28021)
* [[`303a9a3d06`](https://github.com/nodejs/node/commit/303a9a3d06)] - **worker**: make MessagePort constructor non-callable (Anna Henningsen) [#28032](https://github.com/nodejs/node/pull/28032)
* [[`79a8cd0dec`](https://github.com/nodejs/node/commit/79a8cd0dec)] - **worker**: add typechecking for postMessage transfer list (Anna Henningsen) [#28033](https://github.com/nodejs/node/pull/28033)
* [[`d7641d833c`](https://github.com/nodejs/node/commit/d7641d833c)] - **worker**: use DataCloneError for unknown native objects (Anna Henningsen) [#28025](https://github.com/nodejs/node/pull/28025)

targos commented:
We should probably mark the V8 update SEMVER-MINOR in the changelog to avoid confusion

targos commented:
`PR-URL: https://github.com/nodejs/node/pull/28268` in the release commit message

BridgeAR commented:
@nodejs/releasers PTAL. I will update the notable changes with the semver-minor and semver-major PRs. Please point out if something should be marked as notable on top of that.

Ideally, I would love to go into a direction similar to e.g. https://code.visualstudio.com/updates/v1_35 how they point out notable changes. It would be nice to have some pictures or gifs including a small example to newly added features or changed behavior. That would very likely immensely improve the accessibility to our changes.

I will probably not be able to push the release tomorrow (I will be on the plane mainly) but @targos agreed to do that if I do not get to it.

BridgeAR commented:
@bnoordhuis please have a look at https://github.com/nodejs/node/commit/b5dda32b8a. I backported that while creating the release and I am not 100% confident with the way I resolved the conflicts.

targos commented:
The `esm` module is broken by https://github.com/nodejs/node/pull/21387.
I opened https://github.com/standard-things/esm/issues/821 and removed the commit from this release, because this affects multiple module's test suites that depend on `esm`.

targos commented:
I opened https://github.com/nodejs/node/pull/28286 to fix the Windows build

targos commented:
CI is OK (yellow).
Does anyone want to help writing the notable changes?

BridgeAR commented:
I just went through the commits and I came up with this as notable changes:

------

This release contains `semver-major` commits. These are in fact, not "semver-major" due to follow-up patches that remove all breaking changes.

* **build**:
  * The startup time is reduced by enabling V8 snapshots by default #28181
* **deps**:
  * Updated V8 to 7.5.288.22 #27375
  * Updated OpenSSL to 1.1.1c #28211
* **inspector**:
  * The `--inspect-publish-uid` flag was added to specify ways of the inspector web socket url exposure #27741
* **n-api**:
  * Accessors on napi_define_* are now ECMAScript-compliant #27851
* **report**:
  * The cpu info got added to the report output #28188
* **src**:
  * Record the state of the stdio file descriptors on start-up and restore them to that state on exit. This should prevent issues where node.js sometimes leaves stdio in raw or non-blocking mode #24260
* **tools,gyp**:
  * Introduce MSVS 2019 #27375
* **util**:
  * **inspect**:
    * Array grouping got reworked to be more concise and to use a bit more columns than before #28059 #28070
    * Long strings will not be split at 80 characters anymore. Instead they will be split on new lines #28055
* **worker**:
  * `worker.terminate()` now returns a promise and using the callback is deprecated #28021

-----

Two things:
* I did not include the names of the persons involved in the PRs. The names are still next to the commits, so that's hopefully fine?
* I did not word everything properly, so please suggest changes / update the description.

Trott commented:
The OpenSSL update result in a significant (10x) performance degradation for at least one function. I'm guessing that this is acceptably narrow that it won't block the release, but just in case, minimal details are at https://github.com/nodejs/node/issues/28404.

devsnek commented:
Notable change: enables the numeric separator feature by default in v8 (https://v8.dev/features/numeric-separators)

BridgeAR commented:
@nodejs/releasers @nodejs/documentation please have a look at the notable changes.

addaleax commented:
> This release contains `semver-major` commits. These are in fact, not "semver-major" due to follow-up patches that remove all breaking changes.

We should just not label them as semver-major in the changelog, imo… that’s just going to confuse people if we do that

BridgeAR commented:
@addaleax I removed the comment from the notable changes as suggested.

BridgeAR commented:
There was an error while building the Node binaries and I have to rebuild them. @nodejs/releasers it would be good to get a review before the actual release.

targos commented:
Unfortunately the date is wrong in the commit message. I don't know if that's important enough to rebuild everything

BridgeAR commented:
@targos thanks. I just updated it. Since the last build failed and I just restarted a new build just a half an hour before, I stopped the build and build the release with the correct date.

PTAL

targos commented:
LGTM

BridgeAR commented:
@nodejs/build please take a look at the release builds. Two builds failed so far with different targets failing (https://ci-release.nodejs.org/job/iojs+release/4520/ and https://ci-release.nodejs.org/job/iojs+release/4522). The CI seems fine though.

This blocks the current release.

rvagg commented:
first one is just some network connectivity problem on osx

> 07:37:08 ssh: connect to host direct.nodejs.org port 22: Operation timed out

second one is less innocuous on win:

> 13:45:54 MSBUILD : error MSB4166: Child node "2" exited prematurely. Shutting down. Diagnostic information may be found in files in "C:\Users\Administrator\AppData\Local\Temp\1\" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.

but without repeats of this I'd put it down to a temporary glitch on the machine

In this case, where you're having problems that are completely unrelated, just try again. I know it would be nice if it always _just worked_ but there's so many moving parts here that it's not surprising that things occasionally break. But without a pattern that points to something fundamentally broken, just try again and see if a pattern emerges.

I've done a rebuild of 4522's parameters in: https://ci-release.nodejs.org/job/iojs+release/4523/, let's see how that goes.

The other thing to note here is that if you used the same parameters across both runs and your failures were on different targets, you actually got all the built assets in staging that you needed and could go ahead and promote what's there. They should all cover the same release commit. Caution isn't a bad thing of course so if you don't feel confident then holding it up for an all-green is a reasonable approach.

rvagg commented:
https://ci-release.nodejs.org/job/iojs+release/4523/ was all green btw, just run release.sh and it should be good to go
